src/playlist-maker.js

```diff
@@ -133,217 +122,336 @@ const factory = (player, initialList, initialIndex = 0) => {
    * @return {Array}
    *         The playlist
    */
-  const playlist = player.playlist = function(newList, newIndex = 0) {
+  const playlist = player.playlist = (newList, newIndex = 0) => {
     if (Array.isArray(newList)) {
       list = newList.slice();
       if (newIndex !== -1) {
         playlist.currentItem(newIndex);
       }
-      playlist.changeTimeout_ = window.setTimeout(() => {
-        player.trigger('playlistchange');
-      }, 0);
+      player.setTimeout(() => player.trigger('playlistchange'), 1);
     }
 
     // Always return a shallow clone of the playlist list.
     return list.slice();
   };
 
+  // On a new source, if there is no current item, disable auto-advance.
   player.on('loadstart', () => {
     if (playlist.currentItem() === -1) {
       autoadvance.reset(player);
     }
   });
 
-  player.on('dispose', () => {
-    window.clearTimeout(playlist.changeTimeout_);
-  });
+  playlist.currentIndex_ = -1;
+  playlist.player_ = player;
+  playlist.autoadvance_ = {};
+  playlist.repeat_ = false;
 
-  assign(playlist, {
-    currentIndex_: -1,
-    player_: player,
-    autoadvance_: {},
-    repeat_: false,
+  /**
+   * Get or set the current item in the playlist.
+   *
+   * @param  {number} [index]
+   *         If given as a valid value, plays the playlist item at that index.
+   *
+   * @return {number}
+   *         The current item index.
+   */
+  playlist.currentItem = (index) => {
+    if (
+      typeof index === 'number' &&
+      playlist.currentIndex_ !== index &&
+      index >= 0 &&
+      index < list.length
+    ) {
+      playlist.currentIndex_ = index;
+      playItem(
+        playlist.player_,
+        playlist.autoadvance_.delay,
+        list[playlist.currentIndex_]
+      );
```

Comment:
Should this return, so that it works more like vjs getters/setters?

Reply:
Same comment as above, I didn't want to change pre-existing API methods if I could help it.